llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1027,9 +1027,16 @@ static inline bool isAddLike(const SDValue V) {
          (V->getOpcode() == ISD::OR && V->getFlags().hasDisjoint());
 }
 
+static SDValue stripAssertAlign(SDValue N) {
+  while (N.getOpcode() == ISD::AssertAlign)
+    N = N.getOperand(0);
```

Comment:
I think it is probably safe to assume there will only be one of these, so I'd consider removing the while.